tests/istio.mk

```diff
@@ -1,32 +1,61 @@
-# tests/istio.mk defines test targets for Istio
-helm3/test/install:
-	kubectl create ns istio-system || true
-	# Base install with helm3 works only for a fresh cluster - in many cases
-	# we want to upgrade. Helm3 would complain about existing resourceshelm3 install istio-base manifests/charts/charts/base
-	helm3 install -n istio-system istio-16 manifests/charts/charts/istio-control/istio-discovery -f manifests/charts/charts/global.yaml
-	helm3 install -n istio-system istio-canary manifests/charts/charts/istio-control/istio-discovery -f manifests/charts/charts/global.yaml  \
-		--set revision=canary
-
-# Will install or upgrade a 'default' and 'canary' revisions.
-# The canary has DNS capture enabled.
-helm3/test/upgrade:
+
+# Will install or upgrade a 'master' and 'canary' revisions.
+# Use the new style meshConfig: the config map will use defaults+explicit config,
+# no global or values.yaml override
+helm3/base:
 	kubectl create ns istio-system || true
-	helm3 template istio-base manifests/charts/charts/base | kubectl apply -f -
-	helm3 upgrade -i -n istio-system istio-16 manifests/charts/charts/istio-control/istio-discovery \
-		--set global.tag=${TAG} --set global.hub=${HUB} \
-		-f manifests/charts/charts/global.yaml \
-		 --set meshConfig.defaultConfig.proxyMetadata.DNS_CAPTURE="" \
-		 --set meshConfig.defaultConfig.proxyMetadata.DNS_AGENT=""
+	helm3 template --include-crds istio-base manifests/charts/base | kubectl apply -f -
 
+helm3/master:
+	helm3 upgrade -i -n istio-system istio-16 manifests/charts/istio-control/istio-discovery \
+		--set global.tag=${TAG} --set global.hub=${HUB} \
+        --set revision=master \
+		-f manifests/charts/global.yaml \
```

Comment:
What’s your plan for globals.yaml long term?

Reply:
Get rid of them, of course. We're very close - John added 'mesh config' to values.yaml, and we have only 2 charts - istiod and gateways. 

But it's not my top priority- if someone has time, just remove the flag and see what breaks/changes - and move whatever is found to mesh config or values.yaml.